### PR TITLE
Update the changelog to introduce the IoT Plug and Play feature and samples

### PR DESCRIPTION
**Pull request by ahsonkhan**
cc @danewalton, @jspaith, @RickWinter

### CODE REVIEW

**Review by RickWinter — COMMENTED** [review]
Looks good, 
@wduraes do you want anything else in the changelog?

**Review by jspaith — APPROVED** [review]
Thank you for getting this through!

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,11 @@
 
 - Added a current depth field to the JSON reader.
 - Added base64 encoding and decoding APIs that accept `az_span`, available from the `azure/core/az_base64.h` header.
+- Public preview version of a new set of APIs to simplify the experience using IoT Plug and Play. To consume this new feature, a new CMake target has been added `az::iot::hub`. The APIs can be found in the header files: `az_iot_hub_client.h` and `az_iot_hub_client_properties.h`.
```
[the remark this answers is not in the document]
Comment (danewalton) [reply CHANGELOG.md:9]: Is the target new or is it updated?

[the remark this answers is not in the document]
Comment (ahsonkhan) [reply CHANGELOG.md:9]: Good point, it is updated:
https://github.com/Azure/azure-sdk-for-c/blob/c8130121f4c5ca409f58a1a1251cde90c8124564/sdk/src/azure/iot/CMakeLists.txt#L50

**Review by danewalton — APPROVED** [review]

**Pull request merged**